File: Data Structures & Algorithms/is-anagram/submission-8.java
class Solution {
    public boolean isAnagram(String s, String t) {
    if (s.length() != t.length()) return false;
    
    char[] a = s.toCharArray();
    char[] b = t.toCharArray();
    
    for (int i = 0; i < a.length; i++) {
        boolean bulundu = false;
        for (int j = 0; j < b.length; j++) {
            if (a[i] == b[j]) {
                b[j] = '#';  // gerçekten array'i değiştirir
                bulundu = true;
                break;
            }
        }
        if (!bulundu) return false;
    }
    
        return true;
    }
}

File: Data Structures & Algorithms/is-palindrome/submission-8.cs
public class Solution {
    public bool IsPalindrome(string s) {
        char[] ch = s.ToCharArray();
        int a = 0;
        int b = ch.Length - 1;
        while (a < b) {
            while (!char.IsLetterOrDigit(ch[a]) && a < b){
                a++;
            }
            while (!char.IsLetterOrDigit(ch[b]) && a < b){
                b--;
            }
            if ( char.ToLower(ch[a]) != char.ToLower(ch[b]) ){
                return false;
            } else{
                a++;
                b--;
            }
        }
        return true;
    }
}
